### ISSUE TITLE
Build breaks when SCIP is present

### ISSUE THREAD

**Issue by yurivict** [MiniZinc/libminizinc#402]
```
ld: error: duplicate symbol: Algorithms::MinCut::solve()
>>> defined at min_cut.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/lib/algorithms/min_cut.cpp.o:(Algorithms::MinCut::solve())
>>> defined at min_cut.cpp
>>>            CMakeFiles/minizinc_scip.dir/lib/algorithms/min_cut.cpp.o:(.text+0x0)

ld: error: duplicate symbol: MIP_wrapper::getMznLib()
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MIP_wrapper::getMznLib())
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x0)

ld: error: duplicate symbol: MiniZinc::SCIPConstraints::CheckAnnUserCut(MiniZinc::Call const*)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::SCIPConstraints::CheckAnnUserCut(MiniZinc::Call const*))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x30)

ld: error: duplicate symbol: MiniZinc::SCIPConstraints::GetMaskConsType(MiniZinc::Call const*)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::SCIPConstraints::GetMaskConsType(MiniZinc::Call const*))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0xD0)

ld: error: duplicate symbol: MiniZinc::SCIPConstraints::CheckAnnLazyConstraint(MiniZinc::Call const*)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::SCIPConstraints::CheckAnnLazyConstraint(MiniZinc::Call const*))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x80)

ld: error: duplicate symbol: MiniZinc::SECCutGen::print(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::SECCutGen::print(std::__1::basic_ostream<char, std::__1::char_traits<char> >&))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x1C20)

ld: error: duplicate symbol: MiniZinc::SECCutGen::generate(MIP_wrapper::Output const&, std::__1::vector<MIP_wrapper::CutDef, std::__1::allocator<MIP_wrapper::CutDef> >&)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::SECCutGen::generate(MIP_wrapper::Output const&, std::__1::vector<MIP_wrapper::CutDef, std::__1::allocator<MIP_wrapper::CutDef> >&))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x9C0)

ld: error: duplicate symbol: MiniZinc::XBZCutGen::print(std::__1::basic_ostream<char, std::__1::char_traits<char> >&)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::XBZCutGen::print(std::__1::basic_ostream<char, std::__1::char_traits<char> >&))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x4D0)

ld: error: duplicate symbol: MiniZinc::XBZCutGen::generate(MIP_wrapper::Output const&, std::__1::vector<MIP_wrapper::CutDef, std::__1::allocator<MIP_wrapper::CutDef> >&)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::XBZCutGen::generate(MIP_wrapper::Output const&, std::__1::vector<MIP_wrapper::CutDef, std::__1::allocator<MIP_wrapper::CutDef> >&))
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x170)

ld: error: duplicate symbol: MiniZinc::SECCutGen::validate() const
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(MiniZinc::SECCutGen::validate() const)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.text+0x6B0)

ld: error: duplicate symbol: typeinfo for MiniZinc::SECCutGen
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(typeinfo for MiniZinc::SECCutGen)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.data.rel.ro+0x88)

ld: error: duplicate symbol: typeinfo for MiniZinc::XBZCutGen
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(typeinfo for MiniZinc::XBZCutGen)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.data.rel.ro+0x38)

ld: error: duplicate symbol: typeinfo name for MiniZinc::SECCutGen
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(typeinfo name for MiniZinc::SECCutGen)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.rodata+0x16)

ld: error: duplicate symbol: typeinfo name for MiniZinc::XBZCutGen
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(typeinfo name for MiniZinc::XBZCutGen)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.rodata+0x0)

ld: error: duplicate symbol: vtable for MiniZinc::SECCutGen
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(vtable for MiniZinc::SECCutGen)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.data.rel.ro+0x50)

ld: error: duplicate symbol: vtable for MiniZinc::XBZCutGen
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_osicbc.dir/solvers/MIP/MIP_solverinstance.cpp.o:(vtable for MiniZinc::XBZCutGen)
>>> defined at MIP_solverinstance.cpp
>>>            CMakeFiles/minizinc_scip.dir/solvers/MIP/MIP_solverinstance.cpp.o:(.data.rel.ro+0x0)
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

develop branch 01854690be366b964df9e25eaf5bd244759c4e76
FreeBSD 12.1
SCIP-7.0.1

**Comment by Dekker1** [comment]
What commands are you using to compile?

**Comment by yurivict** [comment]
cmake:
```
XDG_DATA_HOME=/usr/ports/math/minizinc/work  XDG_CONFIG_HOME=/usr/ports/math/minizinc/work  HOME=/usr/ports/math/minizinc/work PATH=/usr/ports/math/minizinc/work/.bin:/home/yuri/bin:/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin SHELL=/bin/sh CONFIG_SHELL=/bin/sh \
cmake \
-DUSE_SCIP:BOOL=false -DCMAKE_C_COMPILER:STRING="cc"  -DCMAKE_CXX_COMPILER:STRING="c++"  -DCMAKE_C_FLAGS:STRING="-O2 -pipe -fno-omit-frame-pointer  -fstack-protector-strong -fno-strict-aliasing "  -DCMAKE_C_FLAGS_DEBUG:STRING="-O2 -pipe -fno-omit-frame-pointer  -fstack-protector-strong -fno-strict-aliasing "  -DCMAKE_C_FLAGS_RELEASE:STRING="-O2 -pipe -fno-omit-frame-pointer  -fstack-protector-strong -fno-strict-aliasing "  -DCMAKE_CXX_FLAGS:STRING="-O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer  "  -DCMAKE_CXX_FLAGS_DEBUG:STRING="-O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer  "  -DCMAKE_CXX_FLAGS_RELEASE:STRING="-O2 -pipe -fno-omit-frame-pointer -fstack-protector-strong -fno-strict-aliasing -fno-omit-frame-pointer  "  -DCMAKE_EXE_LINKER_FLAGS:STRING=" -fstack-protector-strong "  -DCMAKE_MODULE_LINKER_FLAGS:STRING=" -fstack-protector-strong "  -DCMAKE_SHARED_LINKER_FLAGS:STRING=" -fstack-protector-strong "  -DCMAKE_INSTALL_PREFIX:PATH="/usr/local"  -DCMAKE_BUILD_TYPE:STRING="Release"  -DTHREADS_HAVE_PTHREAD_ARG:BOOL=YES  -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=YES  -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -GNinja
```

then build with ninja.

**Comment by Dekker1** [comment]
Your commands don't seem complete, but I'm wondering if you can you bring it back to the simplest command for which compilation doesn't work? You are setting a lot of CMake variables, and many should not have anything to do with the problem. If you clone the repository and run:```
mkdir build
cd build
cmake -GNinja ..
cmake --build .
```
do you still have the same problem?

What is the MiniZinc/CMake build configuration output? During the first CMake step you should see something like: 
```
----- MiniZinc build configuration ----
MiniZinc version: 2.4.3 build 175951087
Enabled drivers:
	CPLEX 12.10.0: /builds/minizinc/minizinc/vendor/CPLEX_Studio/cplex/include
	Gecode 6.2.1: /builds/minizinc/minizinc/vendor/gecode/include
	Gurobi 9.0.1: /builds/minizinc/minizinc/vendor/gurobi/include
	OSICBC : /builds/minizinc/minizinc/vendor/cbc/include;/builds/minizinc/minizinc/vendor/cbc/include/coin
	SCIP 6.0.2: /builds/minizinc/minizinc/vendor/scip/include
	XPress: /builds/minizinc/minizinc/vendor/xpressmp/include
---------------------------------------
```

**Comment by yurivict** [comment]
I use the FreeBSD ports framework which sets all these variables. They don't seem to cause any problems, because thousands of projects are built this way.

**Comment by yurivict** [comment]
configure log:
```
-- The CXX compiler identification is Clang 9.0.1
-- The C compiler identification is Clang 9.0.1
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file Windows.h
-- Looking for include file Windows.h - not found
-- Could NOT find CPlex, use CPLEX_ROOT to hint its location (missing: CPLEX_INCLUDE) 
-- Could NOT find Geas, use GEAS_ROOT to hint its location (missing: GEAS_INCLUDE GEAS_LIBRARY) 
-- Found Gecode: /usr/local/include (found suitable version "6.2.0", minimum required is "6.0") found components: Driver Float Int Kernel Minimodel Search Set Support 
-- Could NOT find Gurobi, use GUROBI_ROOT to hint its location (missing: GUROBI_INCLUDE) 
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.11") 
-- Found OsiCBC: /usr/local/include;/usr/local/include/coin (found version "") 
-- Found SCIP: /usr/local/include (found version "7.0.1") 
-- Could NOT find Xpress, use Xpress_ROOT to hint its location (missing: XPRESS_INCLUDE) 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Found Threads: TRUE  
-- Found BISON: /usr/local/bin/bison (found suitable version "3.6.4", minimum required is "3.4") 
-- Found FLEX: /usr/bin/flex (found suitable version "2.5.37", minimum required is "2.5") 

----- MiniZinc build configuration ----
MiniZinc version: 2.4.3 
Enabled drivers:
	Gecode 6.2.0: /usr/local/include
	OSICBC : /usr/local/include;/usr/local/include/coin
	SCIP 7.0.1: /usr/local/include
---------------------------------------

-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_FLAGS_DEBUG
    CMAKE_C_FLAGS_DEBUG
    CMAKE_MODULE_LINKER_FLAGS
    CMAKE_VERBOSE_MAKEFILE
```

**Comment by Dekker1** [comment]
I believe I have found the problem. The MIP solver targets would seem to have shared files in their CMake targets. This doesn't seem to be a problem for a static library (as we normally compile MiniZinc), but since FreeBSD ports sets the option to compile Shared libraries, the linking of these targets fails.

It seems our upstream pipeline is currently stuck because of changes in Chuffed which fail some tests, but I'll merge the fix as soon as possible

**Comment by Dekker1** [comment]
This issue is solved in b350cf37379240a15b176e8429b2dd67f4575dcc
